tap_pendo/streams.py

```diff
@@ -600,19 +600,68 @@ def sync(self, state, start_date=None, key_id=None):
         # Set lookback window
         lookback = bookmark_dttm - timedelta(
             days=self.lookback_window())
-        ts = int(lookback.timestamp()) * 1000
 
+        # get events data
+        events = self.get_events(lookback, state, bookmark_dttm)
+        return (self.stream, events)
+
+    def get_events(self, window_start_date, state, bookmark_dttm):
+        # initialize start date as max bookmark
+        max_bookmark = bookmark_dttm
         # Get period type from config and make request for event's data
         period = self.config.get('period')
-        body = self.get_body(period, ts)
-        events = self.request(self.name, json=body) or []
-        update_currently_syncing(state, None)
-        return self.stream, events
+        # date format to filter
+        date = "date({}, {}, {})"
+        while True:
+            update_currently_syncing(state, self.name)
+
+            # get year, month and day from the start date
+            # if the start date is '2021-01-01' and 'events_date_window' is 25 days
+            # then the window end date will be '2021-01-26'
+            start_date, end_date = round_times(window_start_date, window_start_date + timedelta(days=int(self.config.get('events_date_window', 30))))
```

Comment:
@harshpatel4crest You should also make sure that calculated end date should not be greater than "now()".
Otherwise, your calls would be such that, 
Getting pendo data from "2021-12-09" to "2022-01-09" which is logically not correct.
CC: @savan-chovatiya

Reply:
The Pendo API supports passing future dates. Added a comment in the code for the same.

Reply:
To increase the readability can we take `int(self.config.get('events_date_window', 30))` in one variable? 
And assign it before the loop.

Reply:
Added a variable to store window size.